src/MenuInventory.cpp

```diff
@@ -83,17 +85,52 @@ MenuInventory::MenuInventory()
 				Point pos = Parse::toPoint(infile.val);
 				closeButton->setBasePos(pos.x, pos.y, Utils::ALIGN_TOPLEFT);
 			}
-			// @ATTR equipment_slot|repeatable(int, int, string) : X, Y, Slot Type|Position and item type of an equipment slot.
+			// @ATTR set_button|int, int, filename : Widget X, Widget Y, Image file|Position and image filename for an equipment swap set button.
+			else if(infile.key == "set_button") {
+				int px = Parse::popFirstInt(infile.val);
+				int py = Parse::popFirstInt(infile.val);
+				std::string icon = Parse::popFirstString(infile.val);
+				equipmentSetButton.push_back(new WidgetButton(icon));
+				equipmentSetButton.back()->setBasePos(px, py, Utils::ALIGN_TOPLEFT);
+			}
+			// @ATTR set_previous|int, int, filename : Widget X, Widget Y, Image file|Position and image filename for an equipment swap set previous button.
+			else if(infile.key == "set_previous") {
+				int px = Parse::popFirstInt(infile.val);
+				int py = Parse::popFirstInt(infile.val);
+				std::string icon = Parse::popFirstString(infile.val);
+				equipmentSetPrevious = new WidgetButton(icon);
+				equipmentSetPrevious->setBasePos(px, py, Utils::ALIGN_TOPLEFT);
+			}
+			// @ATTR set_next|int, int, filename : Widget X, Widget Y, Image file|Position and image filename for an equipment swap set next button.
+			else if(infile.key == "set_next") {
+				int px = Parse::popFirstInt(infile.val);
+				int py = Parse::popFirstInt(infile.val);
+				std::string icon = Parse::popFirstString(infile.val);
+				equipmentSetNext = new WidgetButton(icon);
```

Comment:
Allocation of WidgetButtons should happen after all parsing is complete. Store the filenames as strings during parsing and create the buttons once the file is closed. Otherwise, these will leak if the modder redefines `set_next`. This also applies to `set_button` and `set_previous`.

Reply:
Now it's fixed with this commit: ef5368ee6f6f606aa941dd34b628aa05018cc0dc
I chose to declare the raw_ variables that way as they are no longer necessary after the buttons are created/